api.js

```diff
@@ -426,6 +427,64 @@ API.prototype.noticeError = function noticeError(error, customAttributes) {
   this.agent.errors.addUserError(transaction, error, filteredAttributes)
 }
 
+/**
+ * Sends an application log message to New Relic. The agent already
+ * automatically does this for some instrumented logging libraries,
+ * but in case you are using another logging method that is not
+ * already instrumented by the agent, you can use this function
+ * instead.
+ *
+ * If application log forwarding is disabled in the agent
+ * configuration, this function does nothing.
+ *
+ * @param {object} logMessage  The log line object to send.
+ * @param {string} logMessage.message The log message.
+ * @param {string} logMessage.level The log level severity. If this key is
+ *   missing, it will default to UNKNOWN
+ * @param {number} logMessage.timestamp   Unix epoch number denoting the
+ *   time that this log message was produced. If this key is missing,
+ *   it will default to the output of `Date.now()`.
+ */
+API.prototype.recordLogEvent = function recordLogEvent(logMessage) {
+  const metric = this.agent.metrics.getOrCreateMetric(NAMES.SUPPORTABILITY.API + '/recordLogEvent')
+  metric.incrementCallCount()
+
+  if (!applicationLogging.isLogForwardingEnabled(this.agent.config, this.agent)) {
+    logger.warnOnce(
+      'Record logs',
+      'Application log forwarding disabled, method API#recordLogEvent will not record messages'
+    )
+    return
+  }
+
+  // If they don't pass a logMessage object, or it doesn't have the
+  // required `message` key, bail out.
+  if (!logMessage || typeof logMessage !== 'object' || logMessage.message === undefined) {
```

Comment:
you prob want to return all of these `logMessage.<key> === undefined` to `!!logMessage.<key>`. As I'm sure you're learning there are many ways to be falsey and javascript and we don't care which one it is but it needs to be set.

so `logMessage.message = null` or `logMessage.message = undefined` will get hit but `logMessage.message = false` won't

Reply:
I know about falsey, just wanted to explicitly check for missing keys. If they set it to `false` or `''`, I didn't want to catch that here. Also, this will reject `''`, which is currently valid. If you want to log an empty string with `pino` or `winston`, that's gonna get forwarded by us.

I think I'll leave this one as-is but will do the change for timestamp and level.

Reply:
well technically have an empty message in useless in NR logs. so it's ok to reject but now i'm thinking of the case where errors exist, this varies

Reply:
I've decided to change the logic here slightly perhaps more in accord with your suggestions.